src/pages/Graph/SummaryPanelEdge.tsx

```diff
@@ -207,78 +208,108 @@ export default class SummaryPanelEdge extends React.Component<SummaryPanelPropTy
       return;
     }
 
-    const filters = ['request_count', 'request_duration', 'request_error_count', 'tcp_sent', 'tcp_received'];
     const quantiles = ['0.5', '0.95', '0.99'];
     const byLabelsIn = this.getByLabelsIn(sourceMetricType, destMetricType);
 
-    const promise = getNodeMetrics(destMetricType, edge.target(), props, filters, quantiles, byLabelsIn);
-    this.metricsPromise = makeCancelablePromise(promise);
+    const useDest = sourceData.nodeType === NodeType.UNKNOWN || sourceData.namespace === serverConfig().istioNamespace;
+
+    if (!useDest && sourceData.nodeType === NodeType.SERVICE) {
+      // Special case where we must differenciate HTTP metrics from TCP (uses slightly different reporting)
+      const filtersHTTP = ['request_count', 'request_duration', 'request_error_count'];
+      const promiseHTTP = getNodeMetrics(
+        destMetricType,
+        edge.target(),
+        props,
+        filtersHTTP,
+        'inbound',
+        'destination',
+        quantiles,
+        byLabelsIn
+      );
+      const filtersTCP = ['tcp_sent', 'tcp_received'];
+      const promiseTCP = getNodeMetrics(
+        destMetricType,
+        edge.target(),
+        props,
+        filtersTCP,
+        'inbound',
+        'source',
+        quantiles,
+        byLabelsIn
+      );
+      this.metricsPromise = makeCancelablePromise(mergeMetricsResponses([promiseHTTP, promiseTCP]));
+    } else {
+      const filters = ['request_count', 'request_duration', 'request_error_count', 'tcp_sent', 'tcp_received'];
```

Comment:
I'm not sure to understand. Basically, what we do above (in the `if` condition) is split HTTP and TCP queries because they're on different reporters, and here in the `else` they're on the same reporter so we get all at once.

Reply:
Hmmm, this comment actually went with a prior comment that is somehow absent from the review.  Let me restate things here:

I think maybe area of the code  should be broken into two blocks, one for the Http metrics and one for the Tcp metrics.  Above, I don't this useDest has migrated correctly (I could be wrong).  Either way, it may be more clear to have a useDestHttp and useDestTcp, and pull the complementary data using distinct filters.  What do you think?

Reply:
Ok @jshaughn , I've separated the two calls, it's indeed easier to reason about (last commit)